tsn/src/lib.rs

```diff
@@ -151,3 +152,28 @@ pub fn tsn_recv(sock: i32, buf: *mut u8, n: i32) -> isize {
 pub fn tsn_recv_msg(sock: i32, mut msg: msghdr) -> isize {
     unsafe { libc::recvmsg(sock, &mut msg as *mut msghdr, 0) }
 }
+
+pub fn tsn_timespecff_diff(start: &mut TimeSpec, stop: &mut TimeSpec, result: &mut TimeSpec) {
+    if start.tv_sec() > stop.tv_sec()
+        || (start.tv_sec() == stop.tv_sec() && start.tv_nsec() > stop.tv_nsec())
+    {
+        tsn_timespecff_diff(start, stop, result);
+        let result_sec: TimeSpec = TimeValLike::seconds(result.tv_sec());
+        let result_nsec: TimeSpec = TimeValLike::nanoseconds(result.tv_nsec());
+        *result = (result_sec * -1) + result_nsec;
```

Comment:
```suggestion
        *result = -result;
```
https://doc.rust-lang.org/nightly/core/ops/trait.Neg.html#tymethod.neg

Reply:
got an error that says "cannot apply unary operator `-` to type `&mut TimeSpec` cannot apply unary operator `-`"